### PR TITLE
lang/python-iptables: Added package lang/python-iptables

### PR DESCRIPTION
**Pull request by smutt**
Signed-off-by: Andrew McConachie <andrew@depht.com>
Compile tested: AMD64
Run tested: tested
Description: Creation of package lang/python-iptables
https://github.com/ldx/python-iptables

### CODE REVIEW

**Comment by hnyman** [comment]
Same here, too many commits, and also pcapy stuff included in this python-iptables PR.

And you have mixed the git/https logic in PKG_SOURCE_PROTO and PKG_SOURCE_URL options.

One question regarding your patch:
have you tested the package also with LEDE?
I am not sure how the os.uname behaves in the Openwrt derivatives like LEDE. Maybe @commodo knows.
`if os.uname()[1] == 'OpenWrt':`

File: lang/python-iptables/patches/001-find_library.patch [hunk 1]
```diff
@@ -0,0 +1,18 @@
+--- a/iptc/util.py
++++ b/iptc/util.py
+@@ -65,6 +65,15 @@ def _do_find_library(name):
+     except:
+         pass
+ 
++    if os.uname()[1] == 'OpenWrt':
```
Comment (commodo) [inline lang/python-iptables/patches/001-find_library.patch]: i'd say to drop this ; 
not needed ; works without it

File: lang/python-iptables/Makefile [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+PKG_SOURCE_VERSION:=a71f64f40929d934da8189b729dac4cab001083d
+PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_VERSION)
+
+PKG_BUILD_DEPENDS:=python python-setuptools
+
+include $(INCLUDE_DIR)/package.mk
+$(call include_mk, python-package.mk)
+
+define Package/python-iptables
+	SECTION:=language-python
```
Comment (commodo) [inline lang/python-iptables/Makefile]: curios : does this work ?
it's usually `SECTION:=lang`

**Comment by commodo** [comment]
@smutt 
using  `if os.uname()[1] == 'OpenWrt':`  is not recommended

for me, running  `uname -a` gives
```
Linux OpenWrt 4.4.42 #0 SMP Fri Jan 27 14:31:08 2017 ppc GNU/Linux
```
AFAICT , `OpenWrt` is the hostname, not the distro name

example from my host machine
```
sandu@orion  ~/work  uname -a
Linux orion 4.1.36-44-default #1 SMP PREEMPT Mon Dec 12 21:24:10 UTC 2016 (b1c0ac2) x86_64 x86_64 x86_64 GNU/Linux
````

I tried `python-iptables` on LEDE & OpenWrt [there was an email discussion on OpenWrt-devel], and it works fine [without the patch in this PR] ; 
there is an issue tho, that needs a larger discussion involving `ubox`'s kmodloader and Busybox's kmod tools

See here https://github.com/lede-project/source/pull/722

ubox installs  `/usr/sbin/modprobe` 
busybox installs `/sbin/modprobe`

by default, a minimal installation does not have Busybox

@yousong proposed to change the location of ubox's kmodloader, but that's still pending/undecided
 https://patchwork.ozlabs.org/patch/713718/

until then, depending if using `ubox` or `busybox`, you'd need to
`echo /usr/sbin/modprobe > /proc/sys/kernel/modprobe` if using ubox

for Busybox nothing is required

**Comment by yousong** [comment]
> @yousong proposed to change the location of ubox's kmodloader, but that's still pending/undecided
https://patchwork.ozlabs.org/patch/713718/

This is in accepted state and was committed into LEDE master branch in 2017-01-26, the lede-17.01 do not have it though.

**Comment by smutt** [comment]
Thanks everyone for helping me with this. I agree that the "if os.uname()[1] == 'OpenWrt':" line is not needed and will remove it.

The larger problem I've been dealing with is that python-iptables requires shared library support files in /usr/lib/iptables that are not there. I discovered this after starting to test with it and actually build my application. The iptables package builds with core functionality statically linked in the iptables binary and extended functionality(e.g., iptables-mod-filter) available via shared .so placed in /usr/lib/iptables. Package python-iptables needs the core functionality also available via shared library. I've been trying to figure out how to do this but am stuck. Any ideas would be much appreciated. Otherwise I'll mail the package maintainer or dev ML directly.

**Comment by commodo** [comment]
seems you can set a environment variable via  `IPTABLES_LIBDIR`

there's a bit of code in python-iptables
```
        libnames = [name, "lib" + name, name + ext, "lib" + name + ext]
        libdir = os.environ.get('IPTABLES_LIBDIR', None)
        if libdir is not None:
            libdirs = libdir.split(':')
            libs = [os.path.join(*p) for p in product(libdirs, libnames)]
            libs.extend(libnames)
        else:
            libs = libnames
```
if the libdir is specified, it appends the dir as prefix

not sure, if that solves the lib issue

first of all : are you trying this on LEDE or OpenWrt ?
which architecture ?
hopefully i have a similar one

**Comment by smutt** [comment]
I'm using OpenWrt.

There is no problem with python-iptables finding the .so files in libdir. That part works and the directory for OpenWrt is /usr/lib/iptables. The problem is that iptables on OpenWrt is not compiled with all of its functionality in .so files. It is compiled with core functionality statically linked in the iptables binary. So python-iptables can find the libdir, but it doesn't have all the .so files in it. This is probably done to reduce the size of iptables on disk and reduce the amount of memory it takes to run.

If I look at my Debian box /lib/xtables has all the .so files for iptables in it. So python-iptables can just load all of them as needed.

**Comment by smutt** [comment]
Closing this because there are some integration issues with getting this to work on OpenWRT/LEDE that are going to require changing the way iptables is compiled. It doesn't seem worth it to increase the size of the iptables/ip6tables binaries by compiling them with functionality in shared libs (.so) just for this package. And python-iptables requires .so files for all iptables/ip6tables functionality.

**Pull request closed without merging**